metals/src/main/scala/scala/meta/internal/metals/codeactions/FlatMapToForComprehensionCodeAction.scala

```diff
@@ -0,0 +1,541 @@
+            result + newElem
+          )
+
+        (Term.Apply(newFun, newArgs), replacementTimes)
+
+      case Term.ApplyUnary(op, arg) if allowedToGetInsideApply =>
+        val (newArg, argReplacementTimes) =
+          replacePlaceHolder(arg, newName, allowedToGetInsideApply)
+        (Term.ApplyUnary(op, newArg), argReplacementTimes)
+
+      case Term.ApplyUsing(fun, args) if allowedToGetInsideApply =>
+        val (newFun, funReplacementTimes) =
+          replacePlaceHolder(fun, newName, false) // TODO false??
+        val (newArgs, argsReplacementTimes) =
+          args.map(replacePlaceHolder(_, newName, false)).unzip // TODO false??
+
+        val replacementTimes =
+          argsReplacementTimes.fold(funReplacementTimes)((result, newElem) =>
+            result + newElem
+          )
+
+        (Term.ApplyUsing(newFun, newArgs), replacementTimes)
+
+      case Term.ApplyInfix(lhs, op, targs, args) if allowedToGetInsideApply =>
+        val (newLHS, lhsReplacementTimes) =
+          replacePlaceHolder(lhs, newName, allowedToGetInsideApply)
+        val (newArgs, argsReplacementTimes) =
+          args
+            .map(replacePlaceHolder(_, newName, allowedToGetInsideApply))
+            .unzip
+
+        val replacementTimes =
+          argsReplacementTimes.fold(lhsReplacementTimes)((result, newElem) =>
+            result + newElem
+          )
+
+        (Term.ApplyInfix(newLHS, op, targs, newArgs), replacementTimes)
+
+      case Term.Select(qual, name) =>
+        val (newQual, qualReplacementTimes) =
+          replacePlaceHolder(qual, newName, allowedToGetInsideApply)
+        (
+          Term.Select(
+            newQual,
+            name
+          ),
+          qualReplacementTimes
+        )
+      case Term.Placeholder() => (newName, 1)
+      case other => (other, 0)
+    }
+
+  private def replacePlaceHolderInTermWithNewName(
+      term: Term,
+      generatedByMetalsValues: Set[String]
+  ): (Option[(String, Term)], Set[String]) = {
+    val (newName, generatedValues) =
+      createNewName(term, generatedByMetalsValues)
+
+    val (newTerm, replacementTimes) =
+      replacePlaceHolder(term, Term.Name(newName), true)
+    (
+      if (replacementTimes == 1) Some((newName, newTerm)) else None,
+      generatedValues
+    )
+  }
+
+  private def processValueNameAndNextQual(
+      tree: Tree,
+      generatedByMetalsValues: Set[String]
+  ): (Option[(String, Term)], Set[String]) = {
+    tree match {
+      case Term.Function(List(param), term) if param.name.value.isEmpty =>
+        val (newName, generatedValues) =
+          createNewName(tree, generatedByMetalsValues)
+        (Some((newName, term)), generatedValues)
+      case Term.Function(List(param), term) =>
+        (Some((param.name.value, term)), generatedByMetalsValues)
+
+      case Term.AnonymousFunction(term) =>
+        replacePlaceHolderInTermWithNewName(term, generatedByMetalsValues)
+      case term: Term.ApplyInfix =>
```

Comment:
for getting the tests to pass

Reply:
It seems that not at all time, an `AnonymousFunction` is wrapping let’s say an `ApplyInfix` or `ApplyUnary` or the like with placeholders. So in such cases, the execution was falling on the last case which was generating an `Apply` by passing a generated name as an argument to the said Term. Specifically, for example `.map( true && !negate(_) && false)` was failing due to this. So the execution falling on the last case, it was generating `true && !negate(_) && false(generatedByMetals)` instead of `true && !negate(_) && false`
this was happening because the AST Tree corresponding to the expression was not wrapped in `AnonymousFunction`

Reply:
Hmm... that seems like a bug then. Could you make sure that you use the newest scalameta, think something like that was fixed recently. Otherwise we might need to modify the logic a bit - in case of missing `_` placeholder treat is as a normal term.

Reply:
You mean the latest main? I think I rebased on the latest main, just yesterday.

Reply:
`true && !_ && false` is wrapped in `AnonymousFunction` though :|